### PR TITLE
tests: do not run test_switch_provider on CentOS/RHEL 9

### PR DESCRIPTION
**Pull request by ffmancera**
Signed-off-by: Fernando Fernandez Mancera <ffmancera@riseup.net>

### CODE REVIEW

**Comment by richm** [comment]
Are the fedora CI failures known issues?  At least the EL tests look good, with both ansible 2.9 and 2.12

File: tests/tests_switch_provider.yml [hunk 1]
```diff
@@ -6,5 +6,8 @@
 - import_playbook: playbooks/tests_switch_provider.yml
   when:
     # The test requires or should run with NetworkManager, therefore it cannot
-    # run on RHEL/CentOS 6
-    - ansible_distribution_major_version != '6'
+    # run on RHEL/CentOS 6. In addition it cannot run on CentOS/RedHat 9.
+    - (ansible_distribution in ['centOS', 'RedHat'] and
+      ansible_distribution_major_version != '6' and
+      ansible_distribution_major_version | int < 9) or
```
Comment (tyll) [inline tests/tests_switch_provider.yml:12]: IMHO these two when conditions are enouhg (!= 6 and < 9). Restricting to the distribution adds unnecessary complexity IMHO.

**Comment by liangwen12year** [comment]
[citest bad]

File: tests/tests_switch_provider.yml (hunk 1 above)
Comment (ffmancera) [reply tests/tests_switch_provider.yml]: I think you are right. This is fine to me :-)

File: tests/tests_switch_provider.yml [hunk 2]
```diff
@@ -6,5 +6,7 @@
 - import_playbook: playbooks/tests_switch_provider.yml
   when:
     # The test requires or should run with NetworkManager, therefore it cannot
-    # run on RHEL/CentOS 6
-    - ansible_distribution_major_version != '6'
+    # run on RHEL/CentOS 6. In addition it cannot run on CentOS/RedHat 9.
+    - ansible_distribution in ['centOS', 'RedHat'] and
+      ansible_distribution_major_version != '6' and
+      ansible_distribution_major_version | int < 9
```
Comment (tyll) [inline tests/tests_switch_provider.yml:12]: I guess I was not clear enough, my suggestion is this:

```suggestion
    # run on RHEL/CentOS 6 or CentOS/RedHat 9 and later.
    - ansible_distribution_major_version != '6' and
      ansible_distribution_major_version | int < 9
```

**Comment by richm** [comment]
[citest bad]

File: tests/tests_switch_provider.yml (hunk 2 above)
Comment (tyll) [reply tests/tests_switch_provider.yml]: @liangwen12year properly noticed that this change will skip the test also on Fedora, which is wrong. I will revert it.

File: tests/tests_switch_provider.yml [hunk 3]
```diff
@@ -6,5 +6,6 @@
 - import_playbook: playbooks/tests_switch_provider.yml
   when:
     # The test requires or should run with NetworkManager, therefore it cannot
-    # run on RHEL/CentOS 6
-    - ansible_distribution_major_version != '6'
+    # run on RHEL/CentOS 6 or RHEL/CentOS Stream 9 and later.
+    - ansible_distribution_major_version != '6' and
+      ansible_distribution_major_version | int < 9
```
Comment (tyll) [inline tests/tests_switch_provider.yml:11]:
```suggestion
    # run on RHEL/CentOS 6 or RHEL/CentOS Stream 9 and later.
    - ansible_distribution in ['centOS', 'RedHat'] and
      ansible_distribution_major_version != '6' and
      ansible_distribution_major_version | int < 9
```

File: tests/tests_switch_provider.yml [hunk 4]
```diff
@@ -6,5 +6,7 @@
 - import_playbook: playbooks/tests_switch_provider.yml
   when:
     # The test requires or should run with NetworkManager, therefore it cannot
-    # run on RHEL/CentOS 6
-    - ansible_distribution_major_version != '6'
+    # run on RHEL/CentOS 6 or RHEL/CentOS Stream 9 and later.
+    - ansible_distribution in ['centOS', 'RedHat'] and
```
Comment (tyll) [inline tests/tests_switch_provider.yml:10]:
```suggestion
    - ansible_distribution in ['CentOS', 'RedHat'] and
```

**Comment by liangwen12year** [comment]
[citest bad]

**Comment by tyll** [comment]
[citest bad]

**Review by tyll — APPROVED** [review]

**Pull request merged**